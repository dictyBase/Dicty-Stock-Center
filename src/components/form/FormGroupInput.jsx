// @flow
import React, { Component } from 'react'
import { Flex, Box } from 'rebass'
import { ControlLabel, FormGroup, FormControl, HelpBlock } from 'styles'

type Props = {
  field: Object,
  inputType: string,
  placeholder: string
}

export default class FormGroupInput extends Component<Props> {
  displayName = 'form group with input'

  static defaultProps = {
      inputType: 'text'
  }

  render() {
      const { field, inputType, placeholder } = this.props
      const hasError = field.touched && field.error

      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ [1, 1 / 4, 1 / 4] } ml={ 1 } mb={ 2 } mr={ 1 }>
            <ControlLabel>{ this.props.children }</ControlLabel>
          </Box>
          <Box w={ [1, '60%', '60%'] } mr={ 1 } ml={ 1 }>
            <FormControl
              type={ inputType }
              {...field}
              placeholder={ placeholder }
            />
            { hasError && <HelpBlock>{ field.error }</HelpBlock> }
          </Box>
        </Flex>
      </FormGroup>
    )
  }
}
