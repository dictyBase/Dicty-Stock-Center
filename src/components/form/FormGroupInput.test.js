import React from 'react'
import renderer from 'react-test-renderer'
import FormGroupInput from './FormGroupInput'

test('matching a snapshot of FormGroupInput', () => {
    const field = {
        touched: true
    }

    const component = renderer.create(
    <FormGroupInput field={ field } />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
