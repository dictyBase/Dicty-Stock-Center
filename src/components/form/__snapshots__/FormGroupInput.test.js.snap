// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of FormGroupInput 1`] = `
<div
  className="sc-VJcYb eoqXel"
>
  <div
    className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
      is={null}
    >
      <label
        className="sc-fONwsr jwUYlu"
      />
    </div>
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
      is={null}
    >
      <input
        className="sc-ipXKqB dNZJKP"
        placeholder={undefined}
        type="text"
      />
    </div>
  </div>
</div>
`;
